src/main/java/com/oltpbenchmark/benchmarks/auctionmark/AuctionMarkWorker.java

```diff
@@ -447,8 +447,24 @@ protected boolean executeCloseAuctions(Connection conn, CloseAuctions proc) thro
         Timestamp startTime = profile.getLastCloseAuctionsTime();
         Timestamp endTime = profile.updateAndGetLastCloseAuctionsTime();
 
-        List<Object[]> results = proc.run(conn, benchmarkTimes, startTime, endTime);
-
+        List<Object[]> results = null;
+        boolean done = false;
+        // Retry here the close auction procedure if it fails due to a transactional conflict, since
+        // the original submitted procedured has a different type.
+        while (!done) {
+            try {
+                results = proc.run(conn, benchmarkTimes, startTime, endTime);
+                done = true;
+            }
+            catch (SQLException e)  {
+                if (e.getSQLState().startsWith("40")) {
```

Comment:
What is "40"? Is this DBMS specific?

Reply:
SQL states starting with 40 are related to transactional conflicts, such as "serialization failure" and "integrity constraint violation". As far as I know, it is standard across SQL systems.